lib/driver.js

```diff
@@ -238,6 +244,39 @@ class YouiEngineDriver extends BaseDriver {
     
     this.proxydriver = await this.setupNewAndroidDriver(caps);
   }
+  
+  async setupNewMacDriver (caps) {
+    let macArgs = {
+      javascriptEnabled: true
+    };
+    let macdriver = new MacDriver(macArgs);
+    let capsCopy = _.cloneDeep(caps);
+    // Disabling the proxy CommandTimeout in the proxied driver since we are now handling it in the YouiEngine Driver
+    capsCopy.newCommandTimeout = 0;
+
+    await macdriver.createSession(capsCopy);
+
+    return macdriver;
+  }
+
+  async startMacSession (caps) {
+    logger.info("Starting a Mac proxy session");
+    this.proxyAllowList = TO_PROXY_MAC;
+    
+    this.proxydriver = await this.setupNewMacDriver(caps);
+  }
+  
+  startYIMacSession (caps) {  
+    
+    logger.info("Killing app if run");
+    var shell = require('shelljs');
+    var process_name = caps.app.substring(caps.app.lastIndexOf("/") + 1);
+    shell.exec("killall " + process_name);
+    
+    logger.info("Launching app");
+    var child_process = require('child_process'); 
+    child_process.exec(caps.app);
```

Comment:
Neat. You might want to check if `caps.app` exists prior to executing the childprocess and throwing an err incase its not found. Out of curiosity, everytime we instantiate a driver in our `spec_helper` will this also start a YIMacSession? i.e. between every spec when the driver is stopped and restarted, the osx app will be killed and restarted?

Reply:
Yes, that's correct

Reply:
caps.app is a mandatory field

Reply:
actually being checked here: https://github.com/YOU-i-Labs/appium-youiengine-driver/blob/master/lib/driver.js#L175